File: Farmer.dart
class Farmer {
  // Private variables (data hiding)
  double _landArea = 0.0;
  String _cropType = '';

  // Getter methods
  double get landArea => _landArea;
  String get cropType => _cropType;

  // Setter methods
  void setLandArea(double area) {
    if (area >= 0) {
      _landArea = area;
    }
  }

  void setCropType(String crop) {
    if (crop.isNotEmpty) {
      _cropType = crop;
    }
  }
}

// Main function for testing
void main() {
  var farmer = Farmer();
  farmer.setLandArea(10.5);
  farmer.setCropType("Rice");

  print('Land Area: ${farmer.landArea} acres');
  print('Crop Type: ${farmer.cropType}');
}

File: farmer.java
package Encap;

class farmer {
    private double landArea; // ডেটা লুকানো (private)
    private String cropType;

    // মেথডের মাধ্যমে ডেটা অ্যাক্সেস
    public String getCropType() {
        return cropType;
    }

    public void setCropType(String crop) {
        if (crop != null) { // শর্ত যাচাই
            this.cropType = crop;
        }
    }
}